editor/store/selectors.js

```diff
@@ -1196,17 +1200,26 @@ function buildInserterItemFromReusableBlock( enabledBlockTypes, reusableBlock )
  *
  * @param {Object}           state             Global application state.
  * @param {string[]|boolean} enabledBlockTypes Enabled block types, or true/false to enable/disable all types.
+ * @param {Object}           userCapabilities  The current user capabilities.
  *
  * @returns {Editor.InserterItem[]} Items that appear in inserter.
  */
-export function getInserterItems( state, enabledBlockTypes = true ) {
+export function getInserterItems( state, enabledBlockTypes = true, userCapabilities = {} ) {
 	if ( ! enabledBlockTypes ) {
 		return [];
 	}
 
-	const staticItems = getBlockTypes().map( blockType =>
-		buildInserterItemFromBlockType( state, enabledBlockTypes, blockType )
-	);
+	const staticItems = getBlockTypes().map( blockType => {
+		/**
+		 * If the blockType requires a capability and the current user does not have it,
+		 * make sure he cannot use it disabling it.
+		 */
+		if ( userCapabilities && blockType.capability && ! get( userCapabilities, blockType.capability, false ) ) {
+			blockType.isDisabled = true;
```

Comment:
We should be able to avoid this mutating side effect by passing `userCapabilities` into `buildInserterItemFromBlockType` and performing the check there.